liferay-echo/liferay-portal
# LPD-20993 Allow multiple Display Pages and Collections with the same name

## Motivation

Due to this [ticket](https://liferay.atlassian.net/browse/LPD-20993) we decide that we should allow multiple Display Pages and Collections with the same name but only when those elements have a different parent. 

### Changes

Add new finder methods to find by group, name, parentId and type, and delete the older fetchs that now don't return a unique element (G_N_T).

**Also, new test are added to check that we allow this type of duplication and there are no broken tests with this changes.**

## Test scenario 

1. Create a folder (F1) in Display Page Templates.
2. Try to create another one with the same name (you can't).
3. Create a folder (F2) and another into it (F1) (you can because the fist F1 and this one have diferent parents)

Same with Display Page Templates:

1. Create a  Display Page Template in home (DPT1)
2. Create another with the same name into F1 (you can because the fist DPT1 and this one have diferent parents)

For more info see the attached video:

[Screencast from 2024-05-24 13-41-16.webm](https://github.com/liferay-echo/liferay-portal/assets/93203423/d23b76a4-668b-47dc-964a-687e7596a673)
